app/src/main/java/ch/hikemate/app/model/elevation/ElevationServiceRepository.kt

```diff
@@ -0,0 +1,133 @@
+package ch.hikemate.app.model.elevation
+
+import ch.hikemate.app.model.route.LatLong
+import java.io.IOException
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json.Default.decodeFromString
+import kotlinx.serialization.json.buildJsonArray
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.RequestBody
+
+private const val BASE_URL = "https://api.open-elevation.com/api/v1/lookup"
+private const val LOCATION_KEY = "locations"
+private const val LATITUDE_KEY = "latitude"
+private const val LONGITUDE_KEY = "longitude"
+private const val ACCEPT_HEADER = "Accept"
+private const val APPLICATION_JSON = "application/json"
+private const val CONTENT_TYPE_HEADER = "Content-Type"
+private val JSON_MEDIA_TYPE = "application/json; charset=utf-8".toMediaType()
+
+/** Serializable data class for the response from the Elevation API */
+@Serializable data class ElevationResponse(val results: List<ElevationResult>)
+
+/** Serializable data class for the response from the Elevation API */
+@Serializable
+data class ElevationResult(val latitude: Double, val longitude: Double, val elevation: Double)
+
+/**
+ * A repository for the ElevationService. This class is responsible for making the network request
+ */
+class ElevationServiceRepository(private val client: OkHttpClient) : ElevationService {
+
+  // Cache for the elevation data, indexed by hike ID
+  private val cache = mutableMapOf<Long, List<Double>>()
+
+  /**
+   * Get the size of the cache
+   *
+   * @return The size of the cache
+   */
+  fun getCacheSize(): Int {
+    return cache.size
+  }
+
+  override fun getElevation(
+      coordinates: List<LatLong>,
+      hikeID: Long,
+      onSuccess: (List<Double>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    if (coordinates.isEmpty()) {
+      onFailure(Exception("No coordinates provided"))
+      return
+    }
+    // If the cache already has the elevation data for this hike, call onSuccess
+    for (mutableEntry in cache) {
+      if (mutableEntry.key == hikeID) {
+        onSuccess(mutableEntry.value)
+        return
+      }
+    }
+    // Create the JSON body for the request
+    val jsonBody = buildJsonObject {
+      // The major object
+      put(
+          LOCATION_KEY,
+          buildJsonArray {
+            coordinates.forEach { coordinate ->
+              add(
+                  // Object with the location
+                  buildJsonObject {
+                    put(LATITUDE_KEY, coordinate.lat)
+                    put(LONGITUDE_KEY, coordinate.lon)
+                  })
+            }
+          })
+    }
+
+    val body = RequestBody.create(JSON_MEDIA_TYPE, jsonBody.toString())
```

Comment:
Question: Sonar tells us that `create` is deprecated and we shouldn't use it. Is there an alternative?

Reply:
Same as for Rémy, I still need to do some changes specially in the build.gradle. Let me put it on draft for now

Reply:
Solution:
```kotlin
  val jsonBodyString = jsonBody.toString()
  val body: RequestBody = jsonBodyString.toRequestBody(JSON_MEDIA_TYPE)

```
Sorry for this I checked Sonar and forgot to change it.